src/modules/admin/components/forms/FormTreeView.tsx

```diff
@@ -0,0 +1,267 @@
+      return { icon: FormDisplayIcon, text: 'Display', ...defaultQuestion };
+    case ItemType.File:
+      return { icon: FormFileIcon, text: 'File Upload', ...defaultQuestion };
+    case ItemType.Group:
+      return { icon: FormGroupIcon, text: 'Group', ...defaultQuestion };
+    case ItemType.Image:
+      return { icon: FormImageIcon, text: 'Image Upload', ...defaultQuestion };
+    case ItemType.Integer:
+      return { icon: FormIntegerIcon, text: 'Number', ...defaultQuestion };
+    case ItemType.Object:
+      return { icon: FormObjectIcon, text: '', ...defaultQuestion }; // TODO - How should objects be displayed?
+    case ItemType.String:
+      return { icon: FormStringIcon, text: 'Text', ...defaultQuestion };
+    case ItemType.Text:
+      return { icon: FormTextIcon, text: 'Paragraph', ...defaultQuestion };
+    case ItemType.TimeOfDay:
+      return {
+        icon: FormTimeOfDayIcon,
+        text: 'Time of Day',
+        ...defaultQuestion,
+      };
+    default:
+      return { icon: FormObjectIcon, text: '', ...defaultQuestion };
+  }
+};
+
+interface FormTreeLabelProps {
+  children?: React.ReactNode;
+  question?: FormItemDisplay;
+  required?: boolean;
+}
+const FormTreeLabel: React.FC<FormTreeLabelProps> = ({
+  question,
+  children,
+  required,
+  ...other
+}) => {
+  return (
+    <TreeItem2Label
+      {...other}
+      sx={{
+        display: 'flex',
+        alignItems: 'center',
+      }}
+    >
+      {question && (
+        <Badge
+          invisible={!required}
+          badgeContent={'*'}
+          sx={{
+            '& .MuiBadge-badge': {
+              fontWeight: 700,
+              color: theme.palette.error.main,
+              backgroundColor: theme.palette.background.default,
+              borderStyle: 'solid',
+              borderWidth: 1,
+              borderColor: theme.palette.borders.dark,
+              transform: 'scale(1) translate(-30%, -30%)',
+              // TODO: weird visual interaction with required badge when the Tree elements are expanded/contracted
+            },
+          }}
+          anchorOrigin={{
+            vertical: 'top',
+            horizontal: 'left',
+          }}
+        >
+          <Stack
+            sx={{
+              width: 90,
+              alignItems: 'center',
+              color: question.textColor,
+              backgroundColor: question.backgroundColor,
+              borderRadius: '4px 0 0 4px',
+              mr: 2,
+              p: 1, // TODO - This styling is inflexible, looks bad when the label text is long
+            }}
+          >
+            <Box
+              component={question.icon}
+              className='labelIcon'
+              color='inherit'
+              sx={{ fontSize: '1.2rem' }}
+            />
+            <Typography variant='caption'>{question.text}</Typography>
+          </Stack>
+        </Badge>
+      )}
+      {children}
+    </TreeItem2Label>
+  );
+};
+
+interface FormItemDisplay {
```

Comment:
This type could go in a types.ts file for the module if its needed across multiple components / util fns